### PR TITLE
add JSON::Typist::DataPrinter

### PR DESCRIPTION
**Pull request by rjbs**
This is a counterproposal to https://github.com/rjbs/JSON-Typist/pull/4

### CODE REVIEW

File: lib/JSON/Typist/DataPrinter.pm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+=head1 SYNOPSIS
+
+  use JSON::Typist::DataPrinter qw( jdump );
+
+  my $data = get_typed_data_from_your_code();
+
+  say "I got data and here it is!";
+
+  say jdump($data); # ...and you get beautifully printed data
```
Comment (wolfsage) [inline lib/JSON/Typist/DataPrinter.pm:24]: Might be nice to include the beautifully printed data as an example in the pod

File: lib/JSON/Typist/DataPrinter.pm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    $value,
+    (
+      return_value  => 'dump',
+      colored       => 1,
+      show_readonly => 0,
+      filters       => [
+        {
+          'Test::Deep::JType::jstr' => sub { colored([$STRING_COLOR], qq{"$_[0]"}) },
+          'Test::Deep::JType::jnum' => sub { colored([$NUMBER_COLOR], 0 + $_[0]) },
+          'JSON::PP::Boolean'       => sub { colored([$BOOL_COLOR], $_[0] ? 'true' : 'false') },
```
Comment (wolfsage) [inline lib/JSON/Typist/DataPrinter.pm:56]: Oh do we not Test::Deep::JType::jbool or something? wfm

**Review by wolfsage — COMMENTED** [review]
Seems good

File: lib/JSON/Typist/DataPrinter.pm (hunk 1 above)
Comment (rjbs) [reply lib/JSON/Typist/DataPrinter.pm:24]: Right now it's color-based, which means it's not actually really easy to show in Pod.  Probably this needs revising a few times before we're happy with it.  I'll put something in when it's more stable.

**Pull request merged**